server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/MoveExportedFiles.java

```diff
@@ -69,24 +70,27 @@ public Repo<Manager> call(FateId fateId, Manager manager) throws Exception {
 
       Function<FileStatus,String> fileStatusName = fstat -> fstat.getPath().getName();
 
-      Set<String> importing = Arrays.stream(exportedFiles).map(fileStatusName)
-          .map(fileNameMappings::get).collect(Collectors.toSet());
+      Set<Path> importing =
```

Comment:
We are already pushing a bunch of things into the Manager, we should probably be thinking about memory usage and performance. Each of these stream operations is likely creating at least one object which contributes to it being slower than it non-streaming version.

Reply:
> We are already pushing a bunch of things into the Manager, we should probably be thinking about memory usage and performance. Each of these stream operations is likely creating at least one object which contributes to it being slower than it non-streaming version.

Yeah it seems like if the tablet being imported has more files than will fit in memory that you are going to have a bad experience, and streams would definitely exacerbate the problem.  It would probably make sense to open an issue about this bug to lower memory usage.

May also be good  to open another issue about surveying all of the fate operations to look for ones that slurp potentially huge amounts of stuff into memory like this does.  Some of the fate code to avoids slurping too much in memory using [this](https://github.com/apache/accumulo/blob/9d46575487fa2546edfcd55d14f7e91dff626041/core/src/main/java/org/apache/accumulo/core/metadata/schema/Ample.java#L239) and [this](https://github.com/apache/accumulo/blob/9d46575487fa2546edfcd55d14f7e91dff626041/server/manager/src/main/java/org/apache/accumulo/manager/tableOps/bulkVer2/TabletRefresher.java#L81) but I suspect there are other bug out there like this one.  The Guava iterators partition function is really neat and maybe could be used in more situations.

Reply:
Thinking way ahead to one day where Accumulo can use virutal threads and therefore potentially have lots of thread for something like Fate, in that scenario one would need to be much more mindful of memory used per thread.

Reply:
Rather than survey the code to look for things using too much memory, realized it would be better to test it so opened #4654